tests/ui/macros/auxiliary/metavar_2021.rs

```diff
@@ -0,0 +1,16 @@
+//@ edition: 2021
+
+#![feature(macro_metavar_expr)]
+
+#[macro_export]
+macro_rules! make_matcher {
+    ($name:ident, $fragment_type:ident) => {
+        #[macro_export]
+        macro_rules! $name {
+            ($$_:$fragment_type) => { true };
+            ($$($$_:tt)*) => { false };
+        }
+    }
+}
+
+make_matcher!(is_expr_from_2021, expr);
```

Comment:
```suggestion
#[macro_export]
macro_rules! make_matcher {
    ($name:ident, $fragment_type:ident, $d:tt) => {
        #[macro_export]
        macro_rules! $name {
            ($d x:$fragment_type) => { true };
            ($d x:tt { 0 }) => { false };
        }
    };
}

make_matcher!(is_expr_from_2021, expr, $);
```

Perhaps it'd be better to write the test without using the unstable feature, to show how this plausibly affects stable Rust.  We'd expect this to have the same behavior in this test, or no?